scripts/kibana-proxy.py

```diff
@@ -0,0 +1,113 @@
+
+   brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/eb26ed9e7a4b3a35c62a40fa4fec89bf0361781f/Formula/kibana.rb
+
+To use this program, set `AWS_PROFILE` and `source environment`. Then run
+
+   kibana-proxy.py
+
+If multiple versions of Kibana are installed, you may want to select the one to be run by this program by setting
+DSS_KIBANA_BIN (in environment.local) to point at the Kibana executable. For example, I have
+
+   export DSS_KIBANA_BIN=/usr/local/Cellar/kibana/5.5.2/bin/kibana
+"""
+
+import logging
+import shlex
+import signal
+import sys
+from itertools import chain
+
+import boto3
+import os
+
+log = logging.getLogger(__name__)
+
+
+class KibanaProxy:
+
+    def __init__(self, options) -> None:
+        self.options = options
+        self.pids = {}
+
+    def run(self):
+        kibana_port = self.options.kibana_port
+        proxy_port = self.options.proxy_port or kibana_port + 10
+        try:
+            self.spawn('aws-signing-proxy',
+                       '-target', self.dss_end_point,
+                       '-port', str(proxy_port),
+                       AWS_REGION=os.environ['AWS_DEFAULT_REGION'])
+            self.spawn(os.environ.get('DSS_KIBANA_BIN', 'kibana'),
+                       '--port', str(kibana_port),
+                       '--elasticsearch', f'http://localhost:{proxy_port}')
+            self.wait()
+        finally:
+            self.kill()
+
+    @property
+    def dss_end_point(self):
+        log.info('Getting domain endpoint')
+        es = boto3.client('es')
+        domain = es.describe_elasticsearch_domain(DomainName=self.options.domain)
+        return 'https://' + domain['DomainStatus']['Endpoint']
+
+    def spawn(self, *args, **env):
+        command = ' '.join(chain(
+            (k + '=' + shlex.quote(v) for k, v in env.items()),
+            map(shlex.quote, args)))
+        log.info('Running %s', command)
+        pid = os.spawnvpe(os.P_NOWAIT, args[0], args, env={**os.environ, **env})
+        self.pids[pid] = command
+
+    def wait(self):
+        while self.pids:
+            pid, status = os.waitpid(-1, 0)
+            args = self.pids.pop(pid)
+            raise Exception(f'Exited: {args}')
+
+    def kill(self):
+        for pid, args in self.pids.items():
+            log.info('Terminating: %s', args)
+            os.kill(pid, signal.SIGINT)
+            os.waitpid(pid, 0)
+
+
+def main(argv):
+    import argparse
+    cli = argparse.ArgumentParser(description=__doc__)
+    cli.add_argument('--kibana-port', '-p', metavar='PORT', default=5601, type=int,
+                     help="The port Kibana should be listening on.")
+    cli.add_argument('--proxy-port', '-P', metavar='PORT', type=int,
+                     help="The port the proxy should be listening on. The default is the Kibana port plus 10.")
+    default_domain = os.environ.get('DSS_ES_DOMAIN', 'dss-index-' + os.environ['DSS_DEPLOYMENT_STAGE'])
```

Comment:
Why would DSS_ES_DOMAIN not be set?

Reply:
You are right, this should not duplicate the logic for composing the name.